sycl/test-e2e/ESIMD/accessor_global.cpp

```diff
@@ -0,0 +1,121 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// UNSUPPORTED: gpu-intel-gen9 && windows
+//
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// This test verifies usage of accessor methods operator[] and get_pointer().
+
+#include "esimd_test_utils.hpp"
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <iostream>
+
+using namespace sycl;
+using namespace sycl::ext::intel::esimd;
+
+constexpr int VL = 16;
+
+template <typename T, bool TestSubscript>
+bool test(queue Q, uint32_t LocalRange, uint32_t GlobalRange) {
+  std::cout << "Running case: T=" << esimd_test::type_name<T>()
+            << ", TestSubscript=" << TestSubscript << std::endl;
+
+  T *Tmp = new T[GlobalRange * VL];
+  T *Out = malloc_shared<T>(GlobalRange * VL, Q);
+  for (int I = 0; I < GlobalRange * VL; I++)
+    Out[I] = -1;
+
+  try {
+    nd_range<1> NDRange{range<1>{GlobalRange}, range<1>{LocalRange}};
+    buffer<T, 1> TmpBuf(GlobalRange * VL);
+    Q.submit([&](handler &CGH) {
+       auto TmpAcc = TmpBuf.template get_access<access::mode::read_write>(CGH);
+       CGH.parallel_for(NDRange, [=](nd_item<1> Item) SYCL_ESIMD_KERNEL {
+         uint32_t GID = Item.get_global_id(0);
+         uint32_t LID = Item.get_local_id(0);
+         if constexpr (TestSubscript) {
+           for (int I = 0; I < VL; I++)
+             TmpAcc[GID * VL + I] = GID * 100 + I;
+         } else {
+           T *Ptr = TmpAcc.get_pointer();
+           simd<int, VL> IntValues(GID * 100, 1);
+           simd<T, VL> Values = IntValues;
+           block_store(Ptr + GID * VL, Values);
+         }
+
+         Item.barrier();
+
+         if (LID == 0) {
+           for (int LID = 0; LID < LocalRange; LID++) {
+             if constexpr (TestSubscript) {
+               for (int I = 0; I < VL; I++)
+                 Out[(GID + LID) * VL + I] = TmpAcc[(GID + LID) * VL + I];
+             } else {
+               T *Ptr = TmpAcc.get_pointer();
+               simd<T, VL> Values = block_load<T, VL>(Ptr + (GID + LID) * VL);
+               Values.template copy_to(Out + (GID + LID) * VL);
+             }
+           } // end for (int LID = 0; LID < LocalRange; LID++)
+         }   // end if (LID == 0)
+       });
+     }).wait();
+  } catch (sycl::exception const &e) {
+    std::cout << "SYCL exception caught: " << e.what() << '\n';
+    free(Out, Q);
+    return false;
+  }
```

Comment:
I'd rather let the test `std::terminate` by not catching the exception.

Reply:
Thank you for the comment. I'll keep it old-style though, similarly as it is handled in all other tests.